src/libraries/Microsoft.Bcl.Cryptography/src/System/Security/Cryptography/SP800108HmacCounterKdfImplementationManaged.cs

```diff
@@ -0,0 +1,202 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Text;
+
+namespace System.Security.Cryptography
+{
+    internal sealed partial class SP800108HmacCounterKdfImplementationManaged
+    {
+        public SP800108HmacCounterKdfImplementationManaged(ReadOnlySpan<byte> key, HashAlgorithmName hashAlgorithm)
+        {
+            _key = key.ToArray();
+            _hashAlgorithm = hashAlgorithm;
+        }
+
+        internal static unsafe void DeriveBytesOneShot(
+            ReadOnlySpan<byte> key,
+            HashAlgorithmName hashAlgorithm,
+            ReadOnlySpan<byte> label,
+            ReadOnlySpan<byte> context,
+            Span<byte> destination)
+        {
+            if (destination.IsEmpty)
+            {
+                return;
+            }
+
+            byte[] keyBuffer = new byte[key.Length];
```

Comment:
I was about to ask why keyBuffer is using gcnew here.  Then realized it's probably because IncrementalHash wants a precisely-sized array input.  Probably worth a comment.

Reply:
Yep. Added a comment.